pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -522,3 +524,28 @@ func (s *Service) getFilteredSecurityGroupID(securityGroup infrav1.AWSResourceRe
 
 	return *sgs.SecurityGroups[0].GroupId, nil
 }
+
+func getLaunchTemplateInstanceMarketOptionsRequest(spotMarketOptions *infrav1.SpotMarketOptions) *ec2.LaunchTemplateInstanceMarketOptionsRequest {
+	if spotMarketOptions == nil {
+		// Instance is not a Spot instance
+		return nil
+	}
+
+	// Set required values for Spot instances
+	spotOptions := &ec2.LaunchTemplateSpotMarketOptionsRequest{}
+
+	// Persistent option is not applicable EC2 autoscaling. If we do not specify a request type, EC2 makes a one-time request by default.
+	// Persistent requests are only supported when interruption behavior is set to either hibernate or stop.
+	// By default, interruption behavior is `terminate`, so no need to explicitly specify that here as well.
+
+	maxPrice := spotMarketOptions.MaxPrice
+	if maxPrice != nil && *maxPrice != "" {
+		spotOptions.SetMaxPrice(*maxPrice)
+	}
+
+	launchTemplateInstanceMarketOptionsRequest := &ec2.LaunchTemplateInstanceMarketOptionsRequest{}
+	launchTemplateInstanceMarketOptionsRequest.SetMarketType(ec2.MarketTypeSpot)
```

Comment:
I'll be honest, I don't much like the set methods. They are sneaky. :D They return the options, but they operate on a pointer value. But if you don't use the return value, there can be surprises with slices.

I'm not sure about them, they are generated so I usually just stick to applying the value as is to the struct.

This is absolutely a personal opinion. :)

Reply:
I just followed whatever we have today for AWSMachine 😄 although I am ok to use both ways, but can you explain how SET methods breaks rules, would be great if you have some references.

Reply:
Nah, I just wanted to rant a bit. :DD It's fine, since the rest of the whole thing uses it, so you are right that we probably shouldn't mix semantics. :) Otherwise, it's just confusing to which one to use at which point. Also, I know it's convenient because it needs the actual value instead of the pointer.

It's finneeee. Ignore me. :DDD

Reply:
That's completely alright 😄